lib/lumberjack/beats/server.rb

```diff
@@ -139,13 +130,80 @@ def close
       @close.make_true
       @server.close unless @server.closed?
     end
-  end # class Server
+
+    private
+    def retrieve_current_port
+      # Query the port in case the port number is '0'
+      # TCPServer#addr == [ address_family, port, address, address ]
+      @server.addr[1]
+    end
+
+    def certificate_authorities
+      Array(@options[:ssl_certificate_authorities])
+    end
+
+    def server_private_key
+      OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
+                             @options[:ssl_key_passphrase]) if @options[:ssl_key]
+    end
+
+    def server_certificate
+      OpenSSL::X509::Certificate.new(File.read(@options[:ssl_certificate])) if @options[:ssl_certificate]
```

Comment:
See my previous note concerning validation of the options.
https://github.com/logstash-plugins/logstash-input-beats/pull/56#discussion_r56160334